File: .claude/skills/cashu-protocol/scripts/parse_nut.py
#!/usr/bin/env python3
"""Extract sections from Cashu NUT markdown files."""
import sys
import re

def extract_section(file_path, section_name):
    """Find and return a specific section from a NUT file."""
    with open(file_path, 'r') as f:
        content = f.read()

    pattern = rf'^##\s+{re.escape(section_name)}$(.*?)(?=^##\s|\Z)'
    match = re.search(pattern, content, re.MULTILINE | re.DOTALL)

    return match.group(1).strip() if match else None

if __name__ == '__main__':
    if len(sys.argv) != 3:
        print("Usage: parse_nut.py <nut_file> <section_name>")
        sys.exit(1)

    result = extract_section(sys.argv[1], sys.argv[2])
    print(result if result else "Section not found")

File: .claude/skills/cashu-protocol/scripts/parse-nut.ts
#!/usr/bin/env bun
/**
 * Extract sections from Cashu NUT markdown files.
 */

function extractSection(content: string, sectionName: string): string | null {
  const escaped = sectionName.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
  // Match section header, capture until next ## or end of string
  const pattern = new RegExp(`^##\\s+${escaped}\\s*$`, 'm');
  const match = pattern.exec(content);
  if (!match) return null;

  const startIdx = match.index + match[0].length;
  const nextSection = content.slice(startIdx).search(/^##\s/m);
  const endIdx = nextSection === -1 ? content.length : startIdx + nextSection;

  return content.slice(startIdx, endIdx).trim();
}

const args = Bun.argv.slice(2);
if (args.length !== 2) {
  console.log('Usage: bun scripts/parse-nut.ts <nut_file> <section_name>');
  process.exit(1);
}

const [filePath, sectionName] = args;
const content = await Bun.file(filePath).text();
const result = extractSection(content, sectionName);
console.log(result ?? 'Section not found');

export {};
